src/compiler/transformers/es6.ts

```diff
@@ -62,10 +195,19 @@ namespace ts {
             const savedEnclosingBlockScopeContainer = enclosingBlockScopeContainer;
             const savedEnclosingBlockScopeContainerParent = enclosingBlockScopeContainerParent;
 
+            const savedConvertedLoopState = convertedLoopState;
+            if (nodeStartsNewLexicalEnvironment(node) || isClassLike(node)) {

```

Comment:
What about an arrow function in a computed property name or extends clause of a class?